crates/sui-core/src/consensus_handler.rs

```diff
@@ -16,37 +16,42 @@ use tracing::{debug, instrument, warn};
 pub struct ConsensusHandler {
     state: Arc<AuthorityState>,
     sender: mpsc::Sender<ConsensusListenerMessage>,
-    hash: Mutex<u64>,
+    last_seen: Mutex<ExecutionIndicesWithHash>,
 }
 
 impl ConsensusHandler {
     pub fn new(state: Arc<AuthorityState>, sender: mpsc::Sender<ConsensusListenerMessage>) -> Self {
-        let hash = Mutex::new(0);
+        let last_seen = Mutex::new(Default::default());
         Self {
             state,
             sender,
-            hash,
+            last_seen,
         }
     }
 
-    fn update_hash(&self, index: ExecutionIndices, v: &[u8]) -> ExecutionIndicesWithHash {
-        let mut hash_guard = self
-            .hash
+    fn update_hash(&self, index: ExecutionIndices, v: &[u8]) -> Option<ExecutionIndicesWithHash> {
+        let mut last_seen_guard = self
+            .last_seen
             .try_lock()
             .expect("Should not have contention on ExecutionState::update_hash");
+        if last_seen_guard.index <= index {
+            return None;
```

Comment:
Do we not expect the last seen index to be less than the new index, in other words aren't indexes expected to be strictly increasing?

Reply:
Yes indeed, good eye @laura-makdah !

Reply:
Good eye indeed @laura-makdah , totally missed this in review!